File: libcommon/src/main/java/com/soaic/libcommon/base/BasicWebActivity.java
package com.soaic.libcommon.base;

import android.os.Bundle;

import com.soaic.libcommon.R;
import com.soaic.libcommon.weiget.XWebView;

public class BasicWebActivity extends BasicActivity {
    private String curUrl;
    private String curTitle;
    private XWebView webView;
    public static String KEY_URL = "keyUrl";
    public static String KEY_TITLE = "keyTitle";

    @Override protected int getContentView(){
        return R.layout.activity_web;
    }

    @Override protected void initVariables(Bundle savedInstanceState) {
        curUrl = getIntent().getStringExtra(KEY_URL);
        curTitle = getIntent().getStringExtra(KEY_TITLE);
    }

    @Override protected void initViews() {
        webView = findViewById(R.id.webView);
    }

    @Override protected void initEvents() {
        webView.setOnWebViewListener(new XWebView.OnWebViewListener(){

            @Override
            public void onScrollChanged(int dx, int dy) {
            }

            @Override
            public void onProgressChanged(int process) {
                if(process == 100){
                    hideProgressDialog();
                }
            }

            @Override
            public void onTitleChanged(String title) {
            }

            @Override
            public void onUrlChanged(String url) {
            }
        });
    }

    @Override protected void loadData() {
        showProgressDialog();
        webView.loadUrl(curUrl);
    }

    @Override protected void onDestroy() {
        super.onDestroy();
        webView.onDestroy();
    }
}

File: app/src/main/java/com/soaic/toolsapp/ui/activity/base/BasicWebActivity.kt
package com.soaic.toolsapp.ui.activity.base

import android.os.Bundle
import com.soaic.libcommon.weiget.XWebView
import com.soaic.toolsapp.R

class BasicWebActivity : BasicActivity(){

    private lateinit var curUrl : String
    private lateinit var curTitle : String
    private lateinit var webView : XWebView

    companion object {
        const val KEY_URL = "keyUrl"
        const val KEY_TITLE = "keyTitle"
    }

    override val contentView: Int
        get() = R.layout.activity_web

    override fun initVariables(savedInstanceState: Bundle?) {
        curUrl = intent.getStringExtra(KEY_URL)
        curTitle = intent.getStringExtra(KEY_TITLE)
    }

    override fun initViews() {
        webView = findViewById(R.id.webView)
    }

    override fun initEvents() {
        webView.setOnWebViewListener(object : XWebView.OnWebViewListener{
            override fun onScrollChanged(dx: Int, dy: Int) {
            }

            override fun onProgressChanged(process: Int) {
                if(process == 100){
                    hideProgressDialog()
                }
            }

            override fun onTitleChanged(title: String) {

            }

            override fun onUrlChanged(url: String?) {
            }
        })
    }

    override fun loadData() {
        showProgressDialog()
        webView.loadUrl(curUrl)
    }

    override fun onDestroy() {
        super.onDestroy()
        webView.onDestroy()
    }
}